User able to bypass notebooks permissions by using a direct URL

## Problem

In a public project, if a user goes to the launchNotebook URL its able to hit 'Launch server'. For instance: https://renkulab.io/projects/619/launchNotebook

## Effect

The notebook fails to launch because notebooks cannot get the pipelines object (no rights) 

```
[2019-07-07 21:59:09,928] ERROR in notebooks_service: max() arg is an empty sequence
7/7/2019 4:59:10 PM [2019-07-07 21:59:10 +0000] [7] [ERROR] Error handling request /jupyterhub/services/notebooks/cramakrishnantutorial/r10e-ds-py/7c37e099851032b0f3ee9a21f57832498d9a1405
7/7/2019 4:59:10 PM Traceback (most recent call last):
7/7/2019 4:59:10 PM File "/usr/local/lib/python3.7/site-packages/gunicorn/workers/base_async.py", line 56, in handle
7/7/2019 4:59:10 PM self.handle_request(listener_name, req, client, addr)
7/7/2019 4:59:10 PM File "/usr/local/lib/python3.7/site-packages/gunicorn/workers/ggevent.py", line 160, in handle_request
7/7/2019 4:59:10 PM addr)
7/7/2019 4:59:10 PM File "/usr/local/lib/python3.7/site-packages/gunicorn/workers/base_async.py", line 107, in handle_request
7/7/2019 4:59:10 PM respiter = self.wsgi(environ, resp.start_response)
7/7/2019 4:59:10 PM File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 2309, in __call__
7/7/2019 4:59:10 PM return self.wsgi_app(environ, start_response)
7/7/2019 4:59:10 PM File "/app/src/notebooks_service.py", line 113, in __call__
7/7/2019 4:59:10 PM return self.app(environ, start_response)
7/7/2019 4:59:10 PM File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 2295, in wsgi_app
7/7/2019 4:59:10 PM response = self.handle_exception(e)
7/7/2019 4:59:10 PM File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 1741, in handle_exception
7/7/2019 4:59:10 PM reraise(exc_type, exc_value, tb)
7/7/2019 4:59:10 PM File "/usr/local/lib/python3.7/site-packages/flask/_compat.py", line 35, in reraise
7/7/2019 4:59:10 PM raise value
7/7/2019 4:59:10 PM File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 2292, in wsgi_app
7/7/2019 4:59:10 PM response = self.full_dispatch_request()
7/7/2019 4:59:10 PM File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 1815, in full_dispatch_request
7/7/2019 4:59:10 PM rv = self.handle_user_exception(e)
7/7/2019 4:59:10 PM File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 1718, in handle_user_exception
7/7/2019 4:59:10 PM reraise(exc_type, exc_value, tb)
7/7/2019 4:59:10 PM File "/usr/local/lib/python3.7/site-packages/flask/_compat.py", line 35, in reraise
7/7/2019 4:59:10 PM raise value
7/7/2019 4:59:10 PM File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 1813, in full_dispatch_request
7/7/2019 4:59:10 PM rv = self.dispatch_request()
7/7/2019 4:59:10 PM File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 1799, in dispatch_request
7/7/2019 4:59:10 PM return self.view_functions[rule.endpoint](**req.view_args)
7/7/2019 4:59:10 PM File "/app/src/notebooks_service.py", line 169, in decorated
7/7/2019 4:59:10 PM return f(user, *args, **kwargs)
7/7/2019 4:59:10 PM File "/app/src/notebooks_service.py", line 481, in launch_notebook
7/7/2019 4:59:10 PM image = get_notebook_image(user, namespace, project, commit_sha)
7/7/2019 4:59:10 PM File "/app/src/notebooks_service.py", line 369, in get_notebook_image
7/7/2019 4:59:10 PM for pipeline in gl_project.pipelines.list():
7/7/2019 4:59:10 PM AttributeError: 'Response' object has no attribute 'pipelines'
```

## Expected behaviour

The user should get a `You are missing the permissions` message.

![no-permissions-notebook](https://user-images.githubusercontent.com/43743260/60818288-aca5df80-a19d-11e9-8230-eceb756bce53.png)

Or the URL should be invalid.

--- a/src/project/Project.present.js
+++ b/src/project/Project.present.js
@@ -592,43 +592,45 @@ class ProjectViewFiles extends Component {
   }
 }
 
+function notebookLauncher(visibility, notebookLauncher) {
+  let content = null;
+  if (visibility.accessLevel >= ACCESS_LEVELS.DEVELOPER) {
+    content = notebookLauncher;
+  } else {
+    content = (<p>You are missing the permissions to launch Jupyter from this project.</p>);
+  }
+  return (<Col xs={12}>{content}</Col>);
+}
+
 class ProjectNotebookServers extends Component {
   render() {
-    let launch = null;
-    if (this.props.visibility.accessLevel >= ACCESS_LEVELS.DEVELOPER) {
-      launch = <Link to={ `/projects/${this.props.id}/launchNotebook` }>
+    const content = [
+      <Notebooks key="notebooks" standalone={false}
+        projectId={this.props.id} client={this.props.client}
+      />,
+      <Link key="launch" to={ `/projects/${this.props.id}/launchNotebook` }>
         <Button color="primary">Start new server</Button>
       </Link>
-    }
-    else {
-      launch = <p>You are missing the permissions to launch Jupyter from this project.</p>
-    }
+    ];
 
-    return (
-      <Col xs={12}>
-        <Notebooks projectId={this.props.id} client={this.props.client} standalone={false} />
-        {launch}
-      </Col>
-    )
+    return (notebookLauncher(this.props.visibility, content));
   }
 }
 
 class ProjectStartNotebookServer extends Component {
   render() {
-    return (
-      <Col xs={12}>
-        <StartNotebookServer
-          branches={this.props.system.branches}
-          autosaved={this.props.system.autosaved}
-          refreshBranches={this.props.fetchBranches}
-          projectId={this.props.core.id}
-          projectPath={this.props.core.displayId}
-          client={this.props.client}
-          successUrl={this.props.notebookServersUrl}
-          history={this.props.history}
-        />
-      </Col>
-    )
+    let content = (<StartNotebookServer
+      branches={this.props.system.branches}
+      autosaved={this.props.system.autosaved}
+      refreshBranches={this.props.fetchBranches}
+      projectId={this.props.core.id}
+      projectPath={this.props.core.displayId}
+      client={this.props.client}
+      successUrl={this.props.notebookServersUrl}
+      history={this.props.history}
+    />);
+
+    return (notebookLauncher(this.props.visibility, content));
   }
 }
 

